Sources/UtiliKit/ScrollingPageControl/ScrollingPageControl.swift

```diff
@@ -0,0 +1,409 @@
+//
+//  ScrollingPageControl.swift
+//  UtiliKit-iOS
+//
+//  Created by Nathan Chiu on 7/3/19.
+//  Copyright © 2019 Bottle Rocket Studios. All rights reserved.
+//
+
+import UIKit
+
+// MARK: -
+@IBDesignable
+class ScrollingPageControl: UIView {
+    
+    // MARK: Properties
+    
+    // MARK: Modeled Off UIPageControl
+    
+    /// The total number of pages represented, Default is `0`
+    @IBInspectable var numberOfPages: Int {
+        set { _numberOfPages = max(0, newValue) } // prevent a negative number of pages
+        get { return _numberOfPages }
+    }
+    
+    private var _numberOfPages: Int = 0 {
+        didSet {
+            currentPage = _currentPage // keep the current page in range
+            refreshDotLayout()
+        }
+    }
+    
+    /// The index of the currently selected page. Default is `0`. Value pinned to `0..numberOfPages-1`
+    @IBInspectable var currentPage: Int {
+        set {
+            _currentPage = max(0, min(newValue, numberOfPages - 1)) // clamp the newValue between 0 and numberOfPages
+        }
+        get { return _currentPage }
+    }
+    
+    private var _currentPage: Int = 0 {
+        didSet {
+            UIView.animate(withDuration: 0.3, delay: 0.0, usingSpringWithDamping: 1.0, initialSpringVelocity: 0.0, options: .beginFromCurrentState, animations: { [weak self] in
+                self?.updateOffsetIfNeeded()
+                self?.layoutIfNeeded()
+                self?.updateDotColors()
+                self?.didSetCurrentPage?(self?._currentPage ?? 0)
+            })
+        }
+    }
+    
+    /// Hide the the indicator if there is only one page. Default is `false`.
+    @IBInspectable var hidesForSinglePage: Bool = false {
+        didSet { evaluateSelfHiding() }
+    }
+    
+    /// Tint color used to represent pages other than the `currentPage`
+    @IBInspectable var pageIndicatorTintColor: UIColor? = .systemGray {
+        didSet { updateDotColors() }
+    }
+    
+    /// Tint color used to represent the `currentPage`
+    @IBInspectable var currentPageIndicatorTintColor: UIColor? = .systemBlue {
+        didSet { updateDotColors() }
+    }
+    
+    // MARK: Visual Customization
+    /// The number of dots between the margins that will not scale. This cannot be set to a value less than 1.
+    @IBInspectable var mainDotCount: Int {
+        set { _mainDotCount = max(1, newValue) }
+        get { return _mainDotCount }
+    }
+    
+    private var _mainDotCount = 3 {
+        didSet { refreshDotLayout() }
+    }
+    
+    /// The number of dots on each side that will scale if `numberOfPages` is greater than `maxVisibleDots`. This cannot be set to a value less that 0.
```

Comment:
Small typo: "that" --> "than"
```suggestion
    /// The number of dots on each side that will scale if `numberOfPages` is greater than `maxVisibleDots`. This cannot be set to a value less than 0.
```